lib/SILOptimizer/Transforms/SILSROA.cpp

```diff
@@ -219,7 +219,8 @@ createAllocas(llvm::SmallVector<AllocStackInst *, 4> &NewAllocations) {
   }
 }
 
-void SROAMemoryUseAnalyzer::chopUpAlloca(std::vector<AllocStackInst *> &Worklist) {
+void SROAMemoryUseAnalyzer::chopUpAlloca(
+    llvm::SmallVectorImpl<AllocStackInst *> &Worklist) {
   // Create allocations for this instruction.
   llvm::SmallVector<AllocStackInst *, 4> NewAllocations;
   createAllocas(NewAllocations);
```

Comment:
Does this need to propagate 'new allocations' up to DA to ensure its SSA promotion logic handles the newly created allocations?